stores/sql/mysql/main.go

```diff
@@ -53,8 +58,214 @@ func (b *MainDatabase) CreateMigrationTable(ctx context.Context) error {
+	slices := o.Slabs
+	if len(slices) == 0 {
+		return nil // nothing to do
+	}
+
+	usedContracts, err := tx.fetchUsedContracts(ctx, o.Contracts())
+	if err != nil {
+		return fmt.Errorf("failed to fetch used contracts: %w", err)
+	}
+
+	// get contract set id
+	var contractSetID int64
+	if err := tx.QueryRow(ctx, "SELECT id FROM contract_sets WHERE contract_sets.name = ?", contractSet).
+		Scan(&contractSetID); err != nil {
+		return fmt.Errorf("failed to fetch contract set id: %w", err)
+	}
+
+	// insert slabs
+	insertSlabStmt, err := tx.Prepare(ctx, `INSERT INTO slabs (created_at, db_contract_set_id, `+"`key`"+`, min_shards, total_shards)
+						VALUES (?, ?, ?, ?, ?)
+						ON DUPLICATE KEY UPDATE id = last_insert_id(id)`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare statement to insert slab: %w", err)
+	}
+	defer insertSlabStmt.Close()
+
+	querySlabIDStmt, err := tx.Prepare(ctx, "SELECT id FROM slabs WHERE `key` = ?")
+	if err != nil {
+		return fmt.Errorf("failed to prepare statement to query slab id: %w", err)
+	}
+	defer querySlabIDStmt.Close()
+
+	slabIDs := make([]int64, len(slices))
+	for i := range slices {
+		slabKey, err := slices[i].Key.MarshalBinary()
+		if err != nil {
+			return fmt.Errorf("failed to marshal slab key: %w", err)
+		}
+		res, err := insertSlabStmt.Exec(ctx,
+			time.Now(),
+			contractSetID,
+			ssql.SecretKey(slabKey),
+			slices[i].MinShards,
+			uint8(len(slices[i].Shards)),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to insert slab: %w", err)
+		}
+		slabIDs[i], err = res.LastInsertId()
+		if err != nil {
+			return fmt.Errorf("failed to fetch slab id: %w", err)
+		}
+	}
+
+	// insert slices
+	insertSliceStmt, err := tx.Prepare(ctx, `INSERT INTO slices (created_at, db_object_id, object_index, db_multipart_part_id, db_slab_id, offset, length)
+								VALUES (?, ?, ?, ?, ?, ?, ?)`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare statement to insert slice: %w", err)
+	}
+	defer insertSliceStmt.Close()
+
+	for i := range slices {
+		res, err := insertSliceStmt.Exec(ctx,
+			time.Now(),
+			objID,
+			uint(i+1),
+			nil,
+			slabIDs[i],
+			slices[i].Offset,
+			slices[i].Length,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to insert slice: %w", err)
+		} else if n, err := res.RowsAffected(); err != nil {
+			return fmt.Errorf("failed to get rows affected: %w", err)
+		} else if n == 0 {
+			return fmt.Errorf("failed to insert slice: no rows affected")
+		}
+	}
+
+	// insert sectors
+	insertSectorStmt, err := tx.Prepare(ctx, `INSERT INTO sectors (created_at, db_slab_id, slab_index, latest_host, root)
+								VALUES (?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE latest_host = VALUES(latest_host), id = last_insert_id(id)`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare statement to insert sector: %w", err)
+	}
+	defer insertSectorStmt.Close()
+
+	querySectorSlabIDStmt, err := tx.Prepare(ctx, "SELECT db_slab_id FROM sectors WHERE id = last_insert_id()")
```

Comment:
you can avoid calling `last_insert_id()` here because you have it handy at the time you call this statement